src/runtime/contrib/cudnn/conv_forward.cc

```diff
@@ -59,40 +54,125 @@ TVM_REGISTER_GLOBAL("tvm.contrib.cudnn.conv2d.forward")
   // Set Data Type
   entry_ptr->conv_entry.data_type = CuDNNDataType::DLTypeToCuDNNType(String2TVMType(conv_dtype));
   cudnnDataType_t data_type = CuDNNDataType::DLTypeToCuDNNType(x->dtype);
+  // Dims includes N and C
+  int full_dims = dims + 2;
+
+  std::vector<int> dim_v(full_dims);
+  std::vector<int> tensor_stride_v(full_dims);
+
+  // Note: For 2D tenor, using ND setters causes CUDNN_STATUS_NOT_SUPPORTED error
+  // in following cudnnGetConvolutionForwardWorkspaceSize() when data type is fp16, int
+  if (dims == 2) {
   // Set Desc
-  CUDNN_CALL(cudnnSetConvolution2dDescriptor(entry_ptr->conv_entry.conv_desc,
-                                             pad_h,
-                                             pad_w,
-                                             stride_h,
-                                             stride_w,
-                                             dilation_h,
-                                             dilation_w,
-                                             entry_ptr->conv_entry.mode,
-                                             entry_ptr->conv_entry.data_type));
+    CUDNN_CALL(cudnnSetConvolution2dDescriptor(entry_ptr->conv_entry.conv_desc,
+                                               pad_v[0],
+                                               pad_v[1],
+                                               stride_v[0],
+                                               stride_v[1],
+                                               dilation_v[0],
+                                               dilation_v[1],
+                                               entry_ptr->conv_entry.mode,
+                                               entry_ptr->conv_entry.data_type));
+
+    int wn, wc, wh, ww;
+    int xn, xc, xh, xw;
+    int yn, yc, yh, yw;
+    if (entry_ptr->conv_entry.tensor_format ==  CUDNN_TENSOR_NHWC) {
+      wn = static_cast<int>(w->shape[0]);
+      wc = static_cast<int>(w->shape[3]);
+      wh = static_cast<int>(w->shape[1]);
+      ww = static_cast<int>(w->shape[2]);
+
+      xn = static_cast<int>(x->shape[0]);
+      xc = static_cast<int>(x->shape[3]);
+      xh = static_cast<int>(x->shape[1]);
+      xw = static_cast<int>(x->shape[2]);
+
+      yn = static_cast<int>(y->shape[0]);
+      yc = static_cast<int>(y->shape[3]);
+      yh = static_cast<int>(y->shape[1]);
+      yw = static_cast<int>(y->shape[2]);
+    } else {
+      wn = static_cast<int>(w->shape[0]);
+      wc = static_cast<int>(w->shape[1]);
+      wh = static_cast<int>(w->shape[2]);
+      ww = static_cast<int>(w->shape[3]);
+
+      xn = static_cast<int>(x->shape[0]);
+      xc = static_cast<int>(x->shape[1]);
+      xh = static_cast<int>(x->shape[2]);
+      xw = static_cast<int>(x->shape[3]);
+
+      yn = static_cast<int>(y->shape[0]);
+      yc = static_cast<int>(y->shape[1]);
+      yh = static_cast<int>(y->shape[2]);
+      yw = static_cast<int>(y->shape[3]);
+    }
+
```

Comment:
I think you can clean up this, Using variables below

batch_index = 0
channel_index = 1
h_index = 2
w_index = 3
if layout = NHWC
   channel_index = 3
   h_index = 1
   w_index = 2